test/gtest/ternary_tensor_ops.cpp

```diff
@@ -0,0 +1,452 @@
+
+    for(bool packed : packedArr)
+        for(const auto& offsets : offsetsArr)
+        {
+            std::vector<int64_t> final_offsets{0, 0, 0};
+            if(!packed)
+            {
+                if(std::any_of(offsets.begin(), offsets.end(), [](int64_t o) { return o < 0; }))
+                    continue;
+
+                final_offsets = offsets;
+            }
+
+            auto checkStride = [p = packed](const std::vector<int>& lens,
+                                            const std::vector<int>& strides) {
+                if(p)
+                    return true;
+
+                if(lens.size() > strides.size())
+                    return false;
+
+                // only sparsed case allowed, since all the kernels do not support the last
+                // dimension strides
+                if(strides.back() == 1)
+                {
+                    auto packedStrides =
+                        miopen::TensorDescriptor(miopen_type<T>{}, lens).GetStrides();
+                    return std::equal(packedStrides.rbegin(),
+                                      packedStrides.rend(),
+                                      strides.rbegin(),
+                                      [](int ps, int s) { return s >= ps; });
+                }
+
+                // currently tensor operations do not support non-one stride in the last dimention.
+                return false;
+            };
+
+            if(!checkStride(tensorALens, stride_a))
+                continue;
+            if(!checkStride(tensorBLens, stride_b))
+                continue;
+            if(!checkStride(tensorALens, stride_c))
+                continue;
+
+            for(const auto& alphabeta : alphabetaArr)
+                for(const auto& operation : operationArr)
+                {
+                    TestCase& testCase = testCases.emplace_back();
+
+                    testCase.tensorlens_ac = tensorALens;
+                    testCase.tensorlens_b  = tensorBLens;
+                    testCase.alphabeta     = alphabeta;
+                    testCase.offsets       = final_offsets;
+                    testCase.packed        = packed;
+                    testCase.operation     = operation;
+                    testCase.stride_a      = stride_a;
+                    testCase.stride_b      = stride_b;
+                    testCase.stride_c      = stride_c;
+                }
+        }
+}
+
+template <typename T>
+inline auto GetCases()
+{
+    static std::vector<TestCase> testCases;
+
+    if(!testCases.empty())
+    {
+        return testing::ValuesIn(testCases);
+    }
+
+    for(const auto& tensorALens : tensorALensArr)
+        for(const auto& tensorBLens : tensorBLensArr)
+        {
+            if(!checkTensorsCompatibility(tensorALens, tensorBLens))
+            {
+                continue;
+            }
+
+            AddTestCases<T>(testCases, tensorALens, tensorBLens);
+        }
+
+    return testing::ValuesIn(testCases);
```

Comment:
Probably it even should be a `static const auto cases = testing::Values(GenCases());`